chore: add merkle patricia trie diagram

shane-moore opened the pull request:
## Changes
- added diagram for Merkle Patricia Trie to resolve a TODO under `EL > data-structures.md`

shane-moore commented:
hey @taxmeifyoucan, may need some permissions to set you as a reviewer for this PR

raxhvl commented:
Thanks for the PR. The diagram is a little abstract and defines components of Tree than a Patricia Trie.

 A diagram with an example Patricia Trie along with a short explanation could be helpful. [Here is a reference](https://samczsun.com/content/images/2021/05/1920px-Patricia_trie.svg-1-.png)

shane-moore commented:
hey @raxhvl, you made some great points. I refactored to include a Patricia Trie diagram + traversal example.  In addition, I added a section with a Merkle Patricia Trie diagram + traversal example.  Hope these examples help readers get why these data structures were selected in a pre-Verkle world 🌎
